src/core/application/validate-instance-for-update-application-service.ts

```diff
@@ -0,0 +1,215 @@
+}
+
+enum ExecutingAuthorityLevels {
+  LOKAAL = "https://productencatalogus.data.vlaanderen.be/id/concept/UitvoerendBestuursniveau/Lokaal",
+  FEDERAAL = "https://productencatalogus.data.vlaanderen.be/id/concept/UitvoerendBestuursniveau/Federaal",
+  VLAAMS = "https://productencatalogus.data.vlaanderen.be/id/concept/UitvoerendBestuursniveau/Vlaams",
+  EUROPEES = "https://productencatalogus.data.vlaanderen.be/id/concept/UitvoerendBestuursniveau/Europees",
+  PROVINCIAAL = "https://productencatalogus.data.vlaanderen.be/id/concept/UitvoerendBestuursniveau/Provinciaal",
+  DERDEN = "https://productencatalogus.data.vlaanderen.be/id/concept/UitvoerendBestuursniveau/Derden",
+}
+
+
+export class ValidateInstanceForUpdateApplicationService {
+  private readonly _instanceRepository: InstanceRepository;
+  private readonly _bestuurseenheidRepository: BestuurseenheidRepository;
+  private readonly _semanticFormsMapper: SemanticFormsMapper;
+  private readonly _codeRepository: CodeRepository;
+
+  constructor(
+    instanceRepository: InstanceRepository,
+    bestuurseenheidRepository: BestuurseenheidRepository,
+    semanticFormsMapper: SemanticFormsMapper,
+    codeRepository: CodeRepository
+  ) {
+    this._instanceRepository = instanceRepository;
+    this._bestuurseenheidRepository = bestuurseenheidRepository;
+    this._semanticFormsMapper = semanticFormsMapper;
+    this._codeRepository = codeRepository;
+  }
+
+  async validate(
+    bestuurseenheid: Bestuurseenheid,
+    instanceId: Iri,
+    removalsAsTurtleFormat: string,
+    additionsAsTurtleFormat: string
+  ): Promise<ValidationError[]> {
+    const errorList: ValidationError[] = [];
+
+    const currentInstance = await this._instanceRepository.findById(
+      bestuurseenheid,
+      instanceId
+    );
+    const mergedInstance = this._semanticFormsMapper.mergeInstance(
+      bestuurseenheid,
+      currentInstance,
+      removalsAsTurtleFormat,
+      additionsAsTurtleFormat
+    );
+
+    // Validate competent authorities
+    const competentAuthorityErrors = await this.validateAuthoritiesCompetentLevelMapping(mergedInstance.competentAuthorityLevels, mergedInstance.competentAuthorities);
+    if (competentAuthorityErrors){
+      errorList.push(...competentAuthorityErrors);
+    }
+
+    // Validate executing authorities
+    const executingAuthorityErrors = await this.validateAuthoritiesExecutionLevelMapping(
+      mergedInstance.executingAuthorityLevels,
+      mergedInstance.executingAuthorities,
+    );
+    if (executingAuthorityErrors) {
+      errorList.push(...executingAuthorityErrors);
+    }
+
+    return errorList;
+  }
+
+  private async validateAuthoritiesExecutionLevelMapping(
+    selectedLevels: string[],
+    selectedAuthorities: Iri[],
+  ): Promise<ValidationError[]> {
+    const errors: ValidationError[] = [];
+    const authorityLevels = new Set<string>();
+
+    for(const iri of selectedAuthorities){
+      let authorityLevel;
+      if(iri.isOvoCodeIri){
```

Comment:
I'm wondering if we should just always perform a:
        authorityLevel = await this._codeRepository.getExecutionLevel(iri);
No matter if it's ovo or bestuurseenheid?